tests/cover/test_health_checks.py

```diff
@@ -174,6 +174,28 @@ def a(x):
     a()
 
 
+def test_large_base_example_fails_health_check():
+    @given(st.binary(min_size=7000, max_size=7000))
+    def test(b):
+        pass
+
+    with pytest.raises(FailedHealthCheck) as exc:
+        test()
+
+    assert exc.value.health_check == HealthCheck.large_base_example
+
+
+def test_example_that_shrinks_to_overrun_fails_health_check():
+    @given(st.binary(min_size=9000, max_size=9000) | st.none())
```

Comment:
Would `none() | binary(9000, 9000)` also fail?  I tend to think it *should*, because there's a component of the strategy that cannot be drawn from - this is likely to come up fairly regularly in the very large enterprisey strategies. 

On the other hand, digging in to sub-strategies for diagnosis might be too far out of scope here!

Reply:
`none() | binary(9000, 9000)` should fail with a *different* error. It currently doesn't, which is #995.

The thing that's being checked here is specific to the shrinking path, so if you did `none() | binary(9000, 9000)` this would shrink fine to `None`, but would fail when trying to generate examples properly.

Reply:
My proposal is essentially that we should check each branch of the strategy, and fail if any are too large to ever draw from.  That's probably out of scope for this change, but it doesn't seem to fit #995 as written.

Reply:
We don't need to do anything strategy aware for the health checks to catch this example - if deduplication were working correctly this would fail the health check which looks for too many overruns - after you've exhausted the subtree which works (the first branch).

We *could* implement another health check that looks for strategies that we've never seen called successfully, but I think it's a different problem and I'm not sure how widespread this actually is in the wild - the common case I would expect is that people have a lot of strategies that overflow often but not always.

A possibly more fruitful approach is that we do a bunch of random generation that basically inserts this approach on the first draw call to every strategy it sees. We'd need some decently reliable way of distinguishing strategies (I've been vaguely meaning to do something about this with a hashing scheme but haven't worked through the details).

At any rate, I think this is definitely too big a change for now.